tests/light_validation_sync_test.py

```diff
@@ -0,0 +1,112 @@
+
+###############################################################
+# nodeos_chainbase_allocation_test
+#
+# Test snapshot creation and restarting from snapshot
+#
+###############################################################
+
+# Parse command line arguments
+args = TestHelper.parse_args({"-v","--clean-run","--dump-error-details","--leave-running","--keep-logs"})
+Utils.Debug = args.v
+killAll=args.clean_run
+dumpErrorDetails=args.dump_error_details
+dontKill=args.leave_running
+killEosInstances=not dontKill
+killWallet=not dontKill
+keepLogs=args.keep_logs
+
+walletMgr=WalletMgr(True)
+cluster=Cluster(walletd=True)
+cluster.setWalletMgr(walletMgr)
+
+testSuccessful = False
+try:
+    TestHelper.printSystemInfo("BEGIN")
+    cluster.killall(allInstances=killAll)
+    cluster.cleanup()
+
+    # The following is the list of chainbase objects that need to be verified:
+    # - account_object (bootstrap)
+    # - code_object (bootstrap)
+    # - generated_transaction_object
+    # - global_property_object
+    # - key_value_object (bootstrap)
+    # - protocol_state_object (bootstrap)
+    # - permission_object (bootstrap)
+    # The bootstrap process has created account_object and code_object (by uploading the bios contract),
+    # key_value_object (token creation), protocol_state_object (preactivation feature), and permission_object
+    # (automatically taken care by the automatically generated eosio account)
+    assert cluster.launch(
+        pnodes=1,
+        prodCount=1,
+        totalProducers=1,
+        totalNodes=2,
+        useBiosBootFile=False,
+        loadSystemContract=False,
+        specificExtraNodeosArgs={
+            1:"--validation-mode light"})
+
+    producerNode = cluster.getNode(0)
+    validationNode = cluster.getNode(1)
+
+    # Create a transaction to create an account
+    Utils.Print("create a new account payloadless from the producer node")
+    payloadlessAcc = Account("payloadless")
+    payloadlessAcc.ownerPublicKey = "EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV"
+    payloadlessAcc.activePublicKey = "EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV"
+    producerNode.createAccount(payloadlessAcc, cluster.eosioAccount)
+
+
+    contractDir="unittests/test-contracts/payloadless"
+    wasmFile="payloadless.wasm"
+    abiFile="payloadless.abi"
+    Utils.Print("Publish payloadless contract")
+    trans = producerNode.publishContract(payloadlessAcc, contractDir, wasmFile, abiFile, waitForTransBlock=True)
+
+    trx = {
+        "actions": [{"account": "payloadless", "name": "doit", "authorization": [{
+          "actor": "payloadless", "permission": "active"}], "data": ""}],
+        "context_free_actions": [{"account": "payloadless", "name": "doit", "data": ""}],
+        "context_free_data": ["a1b2c3", "1a2b3c"],
+    } 
+
+    cmd = "push transaction '{}' -p payloadless".format(json.dumps(trx))
+    trans = producerNode.processCleosCmd(cmd, cmd, silentErrors=False)
+    assert trans
```

Comment:
Other python tests use `errorExit` shouldn't that be used instead of `assert`?

Reply:
I followed the example from node_chainbase_allocation_test.py which also uses assert. If you like more message to the assertion, I can use the expression "assert condition, message".  The problem of using errorExit() is that you need to use explicit if statement to check the condition which make it more verbose.

Reply:
A message to the assert I think would be helpful.